packages/itwinui-react/src/core/LabeledSelect/LabeledSelect.tsx

```diff
@@ -30,7 +30,7 @@ export type LabeledSelectProps<T> = {
   /**
    * Custom svg icon. Will override status icon if specified.
    */
-  svgIcon?: JSX.Element;
+  svgIcon?: React.ComponentPropsWithoutRef<typeof StatusMessage>['startIcon'];
```

Comment:
we should deprecate this prop to make maintenance easier.

if the user wants to customize the icon, they should pass a `<StatusMessage>` (instead of a string) into `message`

Reply:
also add a patch changeset

Reply:
> we should deprecate this prop to make maintenance easier.
> 
> if the user wants to customize the icon, they should pass a `<StatusMessage>` (instead of a string) into `message`

Sounds good, deprecated `svgIcon` in `LabeledSelect` (https://github.com/iTwin/iTwinUI/pull/1753/commits/a7c897e979e1bcaf34fd270fb7a2bdb699f5e401). Also added a unit test to confirm the replacement we suggested works (https://github.com/iTwin/iTwinUI/pull/1753/commits/223c0b01d79e6c99f61e8c9584dc1d4ba0ec9cb1), and tested the deprecated and replacement code in a ([Stackblitz playground](https://stackblitz.com/edit/github-snplr9?file=src%2FApp.tsx)).

When trying to deprecate `svgIcon` from `InputGroup` also, I noticed that passing `<StatusMessage startIcon={…} />` to the message of `<InputGroup status="…" />` shows duplicate status icons (same playground as above). So, I didn't deprecate `svgIcon` in `InputGroup` and thus kept the `svgIcon` [type correction change](https://github.com/iTwin/iTwinUI/pull/1753/files#diff-eb53afa1cec202f8e446277c6f6ed48b21db0f22f498291fa4cfda68557a4b7dL44-R44) there.

> also add a patch changeset

Since this deprecation is not a bug fix, should it be a minor change? Added a minor changeset for now (https://github.com/iTwin/iTwinUI/pull/1753/commits/bd8470e18b522d78a6ca42eb8d5b92bafae1aab3).

Reply:
> deprecated `svgIcon` in `LabeledSelect` ([a7c897e](https://github.com/iTwin/iTwinUI/commit/a7c897e979e1bcaf34fd270fb7a2bdb699f5e401)). Also added a unit test to confirm the replacement we suggested works ([223c0b0](https://github.com/iTwin/iTwinUI/commit/223c0b01d79e6c99f61e8c9584dc1d4ba0ec9cb1))

The unit test is not needed, because it is testing implementation details of another component. We don't want this test to break if `StatusMessage` changes in the future.

We already tested in [L56](https://github.com/iTwin/iTwinUI/blob/main/packages/itwinui-react/src/core/LabeledSelect/LabeledSelect.test.tsx#L55) that `message` accepts a react node. So anything passed there works unchanged. And we test for `startIcon` in [`StatusMessage.test.tsx`](https://github.com/iTwin/iTwinUI/blob/main/packages/itwinui-react/src/core/StatusMessage/StatusMessage.test.tsx#L40), so this entire scenario is already covered.

> When trying to deprecate `svgIcon` from `InputGroup` also, I noticed that passing `<StatusMessage startIcon={…} />` to the message of `<InputGroup status="…" />` shows duplicate status icons (same playground as above). So, I didn't deprecate `svgIcon` in `InputGroup` and thus kept the `svgIcon` [type correction change](https://github.com/iTwin/iTwinUI/pull/1753/files#diff-eb53afa1cec202f8e446277c6f6ed48b21db0f22f498291fa4cfda68557a4b7dL44-R44) there.

This is a problem. I think we should correctly support passing `string` vs `ReactNode` into `message`. Currently it will put `<StatusMessage>` inside another `<StatusMessage>` if `message={<StatusMessage>}` is used. Let's try to tackle this in a separate PR in the near future. Ideally it would behave similarly to `LabeledSelect`.

> Since this deprecation is not a bug fix, should it be a minor change? Added a minor changeset for now ([bd8470e](https://github.com/iTwin/iTwinUI/commit/bd8470e18b522d78a6ca42eb8d5b92bafae1aab3)).

`minor` is meant for new features, which this is not, so I would go with `patch`. The `svgIcon` prop still works exactly the same as before, so this is not even a significant change.